pkg/imgpkg/signature/cosign.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package signature
+
+import (
+	"fmt"
+	"net/http"
+
+	regname "github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"github.com/k14s/imgpkg/pkg/imgpkg/imageset"
+	"github.com/k14s/imgpkg/pkg/imgpkg/registry"
+)
+
+func NewCosignFinder(reg registry.Registry) *Cosign {
+	return &Cosign{registry: reg}
+}
+
+type Cosign struct {
+	registry registry.Registry
+}
+
+func (c Cosign) Signature(reference regname.Digest) (imageset.UnprocessedImageRef, error) {
+	tag, err := c.signatureTag(reference)
```

Comment:
var name is confusing. it's a "tag" it's a tagRef?

Reply:
done